### DISCUSSION TITLE
A Zipping Gatherer - Gunnar Morling

### DISCUSSION THREAD

**Discussion by giscus** [gunnarmorling/discussions.morling.dev#317] [Announcements]
# A Zipping Gatherer - Gunnar Morling

The other day, I was looking for means of zipping two Java streams:
connecting them element by element—​essentially a join based on stream offset position—​and emitting an output stream with the results.
Unfortunately, there is no zip() method offered by the Java Streams API itself.
While it was considered for inclusion in early preview versions,
the method was removed before the API went GA with Java 8 and you have to resort to 3rd party libraries such as Google Guava if you need this functionality.

https://www.morling.dev/blog/zipping-gatherer/

**Comment by jtama** [comment]
Many thanks for you blog post, that is very clear, and shows a very clever to use of the initializer method.
If I may, I don’t think you should use the ‘ofGreedy’ factory, your integrator is not greedy : i.e it can decide not to take any more element.

**Comment by benhardy** [comment]
Solid post, thanks mate.
